torch/cuda/graphs.py

```diff
@@ -348,19 +369,26 @@ def make_graphed_callables(
     # Capture backward graphs in reverse order
     per_callable_static_grad_outputs = []
     per_callable_static_grad_inputs = []
-    for static_input_surface, static_outputs, bwd_graph, module_params in zip(
+    for static_input_surface, static_outputs, bwd_graph in zip(
         reversed(per_callable_static_input_surfaces),
         reversed(per_callable_static_outputs),
         reversed(bwd_graphs),
-        reversed(per_callable_module_params),
     ):
         # For now, assumes all static_outputs require grad
         # assert all(o.requires_grad for o in static_outputs), "Outputs of graphed callables must require grad."
         static_grad_outputs = tuple(
             torch.empty_like(o) if o.requires_grad else None for o in static_outputs
         )
-
-        with torch.cuda.graph(bwd_graph, pool=mempool):
+        autocast_ctx = (
+            torch.autocast(
+                device_type=static_outputs[0].device.type,
+                enabled=False,
+                cache_enabled=False,
+            )
+            if disable_autocast_bwd and torch.is_autocast_enabled()
+            else nullcontext()
```

Comment:
it's mildly annoying we have to dupe this code twice

Reply:
Refactored to create the context manager with a helper function